src/web/routes/application/check/get-row-data.test.js

```diff
@@ -67,3 +68,83 @@ test('getFlattenedRowData returns flattened row data with step with empty conten
     'should flatten step with content summary and remove step without content summary')
   t.end()
 })
+
+test('groupRowData returns an object with row data grouped by list', (t) => {
+  const rowData = [
+    {
+      list: 'About you',
+      key: 'email address',
+      value: 'my@email.com'
+    },
+    {
+      list: 'About your children',
+      key: 'Do you have children',
+      value: 'yes'
+    },
+    {
+      list: 'About you',
+      key: 'Telephone',
+      value: '111-111-111-111'
+    }
+  ]
+
+  const expected = {
+    'About you': [
+      {
+        list: 'About you',
+        key: 'email address',
+        value: 'my@email.com'
+      },
+      {
+        list: 'About you',
+        key: 'Telephone',
+        value: '111-111-111-111'
+      }
+    ],
+    'About your children': [
+      {
+        list: 'About your children',
+        key: 'Do you have children',
+        value: 'yes'
+      }
+    ]
+  }
+
+  const result = groupRowData(rowData)
+
+  t.deepEqual(result, expected, 'returns an object with row data grouped by list')
+  t.end()
+})
+
+test('getGroupedRowData returns grouped row data', (t) => {
```

Comment:
Feels like this is two tests in one, testing that the grouping by list is performed, and then that if no list key is provided then the default list key will be used. Maybe worth splitting out into two tests?